api/src/main/java/org/openmrs/api/db/ContextDAO.java

```diff
@@ -62,6 +62,29 @@
 	 */
 	public User getUserByUuid(String uuid) throws ContextAuthenticationException;
 	
+	/**
+	 * Gets a user given the username. Privilege checks are not done here because this is used by the
+	 * {@link Context#getAuthenticatedUser()} or {@link Context#authenticate(org.openmrs.api.context.Credentials)} methods.
+	 * 
+	 * @param username The username of the user to fetch
+	 * @return The matched user, null otherwise.
+	 * 
+	 * @since 2.3.0
+	 */
+	public User getUserByUsername(String username);
```

Comment:
Did you get a chance to look at this? https://github.com/openmrs/openmrs-core/blob/master/api/src/main/java/org/openmrs/api/UserService.java